File: Easy/Javascript/268. Missing Number.js
/**Given an array nums containing n distinct numbers in the range [0, n],
return the only number in the range that is missing from the array. */

/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function (nums) {
  let l = nums.length;
  let arr = Array(l + 1).fill(1);

  for (let num of nums) {
    arr[num] = null;
  }

  return arr.findIndex((val) => val != null)
};

console.log(missingNumber([3,0,1]));

File: Easy/Python/268. Missing Number.py
"""Given an array nums containing n distinct numbers in the range [0, n],
return the only number in the range that is missing from the array."""

class Solution:
    def missingNumber(self, nums: list[int]) -> int:
        l = len(nums)
        arr = list(range(l + 1))

        for num in nums:
            arr[num] = None

        return [i for i in arr if i != None][0]

print(Solution().missingNumber([9,6,4,2,3,5,7,0,1]))